### PR TITLE
feat: check application package

### PR DESCRIPTION
**Pull request by demetris-manikas**

### CODE REVIEW

File: src/metadata.ts [hunk 1]
```diff
@@ -24,6 +24,8 @@ export interface AppMetadata extends Metadata {
    */
   readonly description: string
 
+  readonly main?: string
```
Comment (develar) [inline src/metadata.ts]: I think we should not add such option. User should use standard `package.json` main. And, so, our check should check such field in the package.json.

**Comment by develar** [comment]
I think, we should perform this check before pack to asar due to performance reason.  i.e. — when app dir detected, we check it. Before pack, not after.

File: src/metadata.ts (hunk 1 above)
Comment (demetris-manikas) [inline src/metadata.ts]: If I understood correctly interface AppMetadata holds all "useful" properties of the application package.json for the sake ot typesafety. I thought I only gave access to main (an already existent option). Please correct me if I am wrong.

**Comment by demetris-manikas** [comment]
Well, then your comment that I should be aware of asar what was it for?
That is was what made me do the check after the packaging.

File: src/metadata.ts (hunk 1 above)
Comment (develar) [inline src/metadata.ts]: Sorry, you are right.

**Comment by demetris-manikas** [comment]
So should I apply the validation before or after packaging?

**Comment by develar** [comment]
> Well, then your comment that I should be aware of asar what was it for?

Sorry, it was just a proposal. After your changes I see that we can just use app dir.

Also, it will fix OS X — currently, it is broken because on OS X path to `app.asar` is different — `Contents/Resources/app.asar

On other hand — I doubt that it will be a bottleneck (it seems, getting contents of asar file is very quickly). And, strictly speaking, it is better to check final result, not intermediate. 

So, :+1: to current ASAR check, just fix OS X.

File: src/platformPackager.ts [hunk 2]
```diff
@@ -155,14 +157,36 @@ export abstract class PlatformPackager<DC extends PlatformSpecificBuildOptions>
 
     this.beforePack(options)
     await pack(options)
+    await this.sanityCheckPackage(appOutDir, options.asar)
+  }
+
+  private async statPackageFile(appOutDir: string, packageFile: string, isAsar: boolean): Promise<any> {
+    let fpath = path.resolve("/", packageFile)
```
Comment (develar) [inline src/platformPackager.ts]: Please consider to use `const` here.

**Comment by develar** [comment]
Also, please add `asar` dependency explicitly (currently, we inherit it from electron-packager).

**Comment by demetris-manikas** [comment]
I believe all is done.

**Comment by develar** [comment]
Thank you!

**Pull request merged**